File: AudioShopApiService/src/main/java/ru/nonamejack/audioshop/dto/product/ProductDetailsDto.java
package ru.nonamejack.audioshop.dto.product;

import ru.nonamejack.audioshop.dto.attribute.AttributeDto;

import java.math.BigDecimal;
import java.util.List;
import java.util.Map;


public record ProductDetailsDto(
        Integer productId,
        String brandName,
        Integer brandId,
        String categoryName,
        String categoryId,
        String name,
        String description,
        BigDecimal price,
        Integer stock,
        String mainImage,
        Map<Integer, String> imagesMap,
        List<AttributeDto> attributes,
        List<ReviewDto> reviews,
        Boolean active
) {}

File: frontend-admin-panel/src/types/ProductDetailsDto.ts
import type { AttributeDto } from './AttributeDto'
import type { ReviewDto } from './ReviewDto'
export interface ProductDetailsDto {
    productId: number
    brandName: string
    brandId: number
    categoryName: string
    categoryId: number
    name: string
    description?: string
    price: number
    stock: number
    mainImage?: string
    imagesMap: Record<number, string>
    attributes: AttributeDto[]
    reviews: ReviewDto[]
    active: boolean
}